llvm/lib/Target/AMDGPU/AMDGPUInsertSingleUseVDST.cpp

```diff
@@ -132,6 +132,153 @@ class AMDGPUInsertSingleUseVDST : public MachineFunctionPass {
 
   AMDGPUInsertSingleUseVDST() : MachineFunctionPass(ID) {}
 
+  static bool IsValidOpcode(const MachineInstr &MI) {
```

Comment:
This feels like it needs to be an instruction bit and/or searchable table

Reply:
Or could we derive this from the scheduling info?

Reply:
Searchable table sounds good. We can't derive it from the sched model because it's not really related to how many cycles the instruction takes to execute; we should treat it as a separate list of instructions defined by the hardware design.

Reply:
I've changed this to be a searchable table.